privacyidea/app.py

```diff
@@ -218,18 +232,53 @@ def create_app(config_name="development",
         DEFAULT_LOGGING_CONFIG["loggers"]["privacyidea"]["level"] = level
         logging.config.dictConfig(DEFAULT_LOGGING_CONFIG)
 
-    babel = Babel(app)
-
-    @babel.localeselector
-    def get_locale():
-        return get_accepted_language(request)
-
     queue.register_app(app)
 
-    if init_hsm:
+    if initialize_hsm:
         with app.app_context():
             init_hsm()
 
+    if config_name == "testing":
+        # we are running the tests, create the database
+        with app.app_context():
+            db.create_all()
+
+    # check that we have a correct node_name -> UUID relation
+    with app.app_context():
+        # first check if we have a UUID in the config file which takes precedence
+        pi_uuid = app.config.get("PI_UUID")
+        if not pi_uuid:
+            # we try to get the unique installation id (See <https://0pointer.de/blog/projects/ids.html>)
+            try:
+                with open('/etc/machine-id', 'r') as f:
+                    pi_uuid = uuid.UUID(f.read().strip())
+            except Exception as e:  # pragma: no cover
+                logging.getLogger(__name__).info(f"Could not determine the machine id: {e}")
+            else:
+                # we generate a random UUID which will change on every startup
+                pi_uuid = uuid.uuid4()
+                logging.getLogger(__name__).warning("Generating a random UUID! "
+                                                    "If persisting the UUID fails, "
+                                                    "it will change on every application start")
+            app.config["PI_UUID"] = str(pi_uuid)
+            # safe UUID to config file
+            try:
+                with open(config_file, 'a') as f:  # pragma: no cover
+                    f.write(f"PI_UUID = \"{str(pi_uuid)}\"\n")
+            except IOError as exx:
+                logging.getLogger(__name__).warning(f"Could not add UUID to config "
+                                                    f" file '{config_file}': {exx}")
+
+        pi_node_name = app.config.get("PI_NODE") or app.config.get("PI_AUDIT_SERVERNAME", "localnode")
+        insp = sa.inspect(db.get_engine())
```

Comment:
Ah, I understand we need to do this, since the ORM is not ready at this stage, right?

Reply:
if it is only for non-run migrations, please remove this.